kura/org.eclipse.kura.linux.watchdog/src/main/java/org/eclipse/kura/linux/watchdog/WatchdogServiceImpl.java

```diff
@@ -63,34 +57,57 @@ protected void activate(Map<String, Object> properties) {
     protected void deactivate() {
         cancelPollTask();
         shutdownPollExecutor();
-        if (this.options.isEnabled()) {
+        if (this.watchdogFileWriter != null) {
             refreshWatchdog();
+            closeWatchdogFileWriter();
         }
-        closeWatchdogFileWriter();
     }
 
     public void updated(Map<String, Object> properties) {
+        WatchdogServiceOptions newOptions = new WatchdogServiceOptions(properties);
+
         cancelPollTask();
+        this.pollExecutor.submit(() -> {
+            Thread.currentThread().setName("WatchdogServiceImpl");
+            if (this.watchdogFileWriter != null) {
```

Comment:
It's not enough to just check for not null - it may have been enabled and disabled before and when you only modify configuration of a disabled service, write will fail with exception:
```
Failed to write magic character to watchdog device '/dev/watchdog'
java.io.IOException: Stream closed
```